### ISSUE TITLE
Discuss: Assets only support Object (should support string shorthand)

### ISSUE THREAD

**Issue by hansl** [angular/angular-cli#10270]
Corollary to https://github.com/angular/angular-cli/issues/10267

**Comment by filipesilva** [comment]
So for this to happen we'd have to look at
```
assets: [
  "src/favicon.ico",
  "src/assets",
]
```

and know that it should be converted to
```
assets: [
  {
    "glob": "src/favicon.ico",
    "input": "src/",
    "output": "/"
  },
  {
    "glob": "**/*",
    "input": "src/assets",
    "output": "/assets"
  }
]
```

There's two parts to this:
- differentiating between directories (glob is `**/*`) and files (glob is the file itself)
- determining output paths

It's fairly straightforward to determine the glob by checking the if the string is for a dir or for a file.

But it's not easy at all to determine the output path. We usually put stuff in `src/`, and expect `src/` to 'translate' to the output root. But this concept is not explicit or configurable.

In other words: just by looking at `"src/assets"` we cannot know that the output path should be `/assets` because we don't know that `src/` is conceptual output root.

**Comment by filipesilva** [comment]
We addressed this issue by adding a `sourceRoot` property to projects that is meant to represent the root of the source files, assets and index.html file structure.
